### PR TITLE
A tiny change: use a phylogenetic directory

### PR DESCRIPTION
**Pull request by j23414**
## Description of proposed changes

A tiny change ;) Use a phylogenetic directory to follow the pathogen-repo-template.

* https://github.com/nextstrain/pathogen-repo-template

## Checklist

- [ ] Checks pass
- [ ] Manual check passes

```
git clone https://github.com/nextstrain/dengue.git
cd dengue
git checkout phylogenetic_dir
cd phylogenetic
nextstrain build . 
```

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -6,4 +6,6 @@ on:
 
 jobs:
   ci:
-    uses: nextstrain/.github/.github/workflows/pathogen-repo-ci.yaml@master
+    uses: nextstrain/.github/.github/workflows/pathogen-repo-ci.yaml@dec0880059017dac7facf100435c5737bf1386c8
+    with:
+      workflow-root: phylogenetic
```
Comment (joverlee521) [inline .github/workflows/ci.yaml:11]: This workflow does not fully work as intended (see [Tom's comment in original PR](https://github.com/nextstrain/.github/pull/57#discussion_r1336389847)).

You may want to use the stopgap I've used in mpox (https://github.com/nextstrain/mpox/pull/214) until the pathogen-repo-ci is updated.

File: README.md [hunk 2]
```diff
@@ -1,78 +1,14 @@
-# nextstrain.org/dengue
+# Nextstrain repository for dengue virus
 
-This is the [Nextstrain](https://nextstrain.org) build for dengue, visible at
-[nextstrain.org/dengue](https://nextstrain.org/dengue).
+[![pre-commit.ci status](https://results.pre-commit.ci/badge/github/nextstrain/dengue/main.svg)](https://results.pre-commit.ci/latest/github/nextstrain/dengue/main)
```
Comment (joverlee521) [inline README.md:3]: Since we are not using pre-commit CI in this repo, I recommend removing this to prevent any confusion.

**Review by joverlee521 — CHANGES_REQUESTED** [review]
The majority of the changes look good to me 👍 

My only request to is update the CI workflow since it currently does not fully work as intended. Looking at the [latest CI job on this branch](https://github.com/nextstrain/dengue/actions/runs/7332417961), there are no artifacts uploaded due to the bug so we cannot inspect the build outputs.

File: README.md (hunk 2 above)
Comment (j23414) [reply README.md]: Dropped with commit https://github.com/nextstrain/dengue/pull/15/commits/71a332c71f96d1610b54fc119385f14b8b24aa40

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (j23414) [reply .github/workflows/ci.yaml]: Thanks for the explanation! Incorporated changes in commit https://github.com/nextstrain/dengue/pull/15/commits/79d1be45d3917f8350d25585beb2eeff625a3cd4

**Comment by j23414** [comment]
Hi @joverlee521! This PR is ready for review. Following our earlier discussion, I've implemented further changes, but am open to further revisions.

* Use decompressed example files
* Refactor and relocate certain dengue-specific functions from the `phylogenetic/Snakefile` to a more generalized `dengue_config.yaml`

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -5,7 +5,17 @@ on:
...
+    uses: nextstrain/.github/.github/workflows/pathogen-repo-build.yaml@master
+    secrets: inherit
     with:
-      workflow-root: phylogenetic
\ No newline at end of file
+      runtime: ${{ matrix.runtime }}
+      run: |
+        nextstrain build \
+          phylogenetic \
+          --configfile profiles/ci/profiles_config.yaml
```
Comment (joverlee521) [inline .github/workflows/ci.yaml:21]: Ah, looks like we missed a [couple lines](https://github.com/nextstrain/mpox/blob/0ff5ce863c1c8c51e0daf164bfbba54d5b7fef24/.github/workflows/ci.yaml#L25-L31) when we were copying over the CI workflow from mpox

```suggestion
          --configfile profiles/ci/profiles_config.yaml
      artifact-name: output-${{ matrix.runtime }}
      artifact-paths: |
        phylogenetic/auspice/
        phylogenetic/results/
        phylogenetic/benchmarks/
        phylogenetic/logs/
        phylogenetic/.snakemake/log/
```

The `artifact-name` ensures that we produce separate artifacts for the docker and conda builds
The `artifact-paths` is used to specify paths within the phylogenetic directory.

File: phylogenetic/README.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+The above datasets have been preprocessed and cleaned from GenBank and are updated at regular intervals. 
+
+### Using example data
+
+Alternatively, you can run the build using the
+example data provided in this repository.  Before running the build, copy the
+example sequences into the `data/` directory like so:
+
+    mkdir -p data/
+    cp example_data/dengue* data/
```
Comment (joverlee521) [inline phylogenetic/README.md:59]: One other benefit to having the CI profile is users no longer have to manually copy over the example data.
These instructions can be updated to use
```
nextstrain build .  --configfile profiles/ci/profiles_config.yaml
```

**Review by joverlee521 — APPROVED** [review]
Thanks for working through this! I caught a couple smaller things but should be good to merge!

**Pull request merged**